### PR TITLE
Updating rh-che to the latest 6.16.0 upstream version

### PR DESCRIPTION
**Pull request by ibuziuk**
### What does this PR do?
Updating rh-che to the latest 6.16.0 upstream version

### What issues does this PR fix or reference?
N/A

### How have you tested this PR?
Not tested

### CODE REVIEW

**Comment by ibuziuk** [comment]
[test]

**Comment by garagatyi** [comment]
@ibuziuk should this PR also include https://github.com/redhat-developer/rh-che/pull/1130?

**Comment by ibuziuk** [comment]
@garagatyi if it is not directly related to `6.16.0` I would prefer to merge it separately

**Comment by garagatyi** [comment]
@ibuziuk as far as I understand binding is supposed to fail without that PR. But it works \o/

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
workspace startup is failing due to:

> 2019-01-02 11:44:15,251[rcherInitThread]  [INFO ] [o.e.c.a.s.s.i.LuceneSearcher 159]    - Initial indexing complete after 3 msec 
> 	at org.apache.catalina.filters.CorsFilter.parseAndStore(CorsFilter.java:786)
> javax.servlet.ServletException: It is not allowed to configure supportsCredentials=[true] when allowedOrigins=[*]
> 	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:266)
> 	at com.google.inject.servlet.GuiceFilter.init(GuiceFilter.java:232

Need to remove `CHE_WSAGENT_CORS_ALLOWED__ORIGINS` from the template (it will be automatically detected and set based on workspace master domain)

**Comment by ibuziuk** [comment]
@garagatyi @amisevsk @davidfestal could you please approve ?

File: openshift/rh-che.config.yaml [hunk 1]
```diff
@@ -41,7 +41,6 @@ data:
   che-system-admin-name: "admin"
   che-system-super-privileged-mode: "true"
   che-websocket-endpoint: "wss://che.prod-preview.openshift.io/api/websocket"
-  che-wsagent-cors-allowed-origins: "*"
```
Comment (garagatyi) [inline openshift/rh-che.config.yaml]: Have we finished checks that everything works without CORS?

Comment (rhopp) [reply openshift/rh-che.config.yaml]: Katka done some testing: https://github.com/redhat-developer/rh-che/issues/1116#issuecomment-446896060

Comment (garagatyi) [reply openshift/rh-che.config.yaml]: Thanks for the details!

**Comment by ibuziuk** [comment]
related HK issue for removing `che-wsagent-cors-allowed-origin` cm entry https://gitlab.cee.redhat.com/dtsd/housekeeping/issues/2498

**Pull request merged**